tests/fs/test_trackingdelegate.c

```diff
@@ -24,23 +35,86 @@ int main() {
     FS.trackingDelegate['onDeletePath'] = function(path) {
       out('Deleted "' + path + '"');
     };
-    FS.trackingDelegate['onOpenFile'] = function(path, flags) { 
-      out('Opened "' + path + '" with flags ' + flags);
+    FS.trackingDelegate['onOpenFile'] = function(path, flags) {
+      var stat = FS.stat(path);
+      var fileSize = stat.size;
+      if (flags & O_CREAT) {
+        flags &= O_CREAT;
+      } else if (flags & O_TRUNC) {
+        flags &= O_TRUNC;
+      } else if (flags & O_APPEND) {
+        flags &= O_APPEND;
+      } else if (flags & O_EXCL) {
+        flags &= O_EXCL;
+      } else if (flags & O_WRONLY) {
+        flags &= O_WRONLY;
+      } else if (flags & O_RDWR) {
+        flags &= O_RDWR;
+      // O_RDONLY = 32768
```

Comment:
This flags logic doesn't look quite right. It looks like `O_TRUNC | O_WRONLY`, for example, would be transformed into just `O_TRUNC`.

Reply:
Yep, good point. I updated it to just use an array for multiple modes. 
https://man7.org/linux/man-pages/man2/open.2.html